service/frontend/api/domain_handlers_test.go

```diff
@@ -278,3 +280,140 @@ func TestDescribeDomain(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteDomain(t *testing.T) {
+	domainName := "domain-name"
+	testCases := []struct {
+		name          string
+		req           *types.DeleteDomainRequest
+		setupMocks    func(*mockDeps)
+		expectError   bool
+		expectedError string
+	}{
+		{
+			name: "success - delete deprecated domain",
```

Comment:
nit:
It is better to call subtests with underscope:
success_delete_deprecated_domain

This way, when a test fails, you can easily find the failed case.
When you use spaces and dashes, it will not be apparent which test case caused the failure.
And I think it will look like:
TestDeleteDomain/success_-_delete_deprecated_domain